### PR TITLE
DPS: update client library documentation

### PR DESCRIPTION
**Pull request by Danielius1922**
## Summary by CodeRabbit

- **New Features**
    - Introduced new functions for handling multiple DPS server endpoints.
    - Added verification for DPS endpoint certificate's Common Name to match the selected endpoint.
    - Enhanced cloud status observer mechanisms and introduced custom retry actions for improved reliability.
- **Documentation**
    - Expanded documentation on provisioning status values and DPS configuration.
- **Deprecations**
    - Deprecated the function for setting a single DPS server endpoint.

### CODE REVIEW

File: content/en/docs/services/device-provisioning-service/retry-mechanism.md [hunk 1]
```diff
@@ -39,14 +41,41 @@ Non-transient errors force a full provisioning on retry. Transient errors first
...
+```pseudocode
+
+  timeout = take the timeout value from configuration array indexed by the retry counter
+
+  delay = timeout * / 2;
+  // Include a random delay to prevent multiple devices from attempting to
+  // connect or make requests simultaneously.
+  delay += random value % delay;
+
+```
```
[the remark this answers is not in the document]
Comment (Danielius1922) [reply content/en/docs/services/device-provisioning-service/retry-mechanism.md:57]: @coderabbitai, what implementation would you suggest (written in C)

File: content/en/docs/services/device-provisioning-service/features.md [hunk 2]
```diff
@@ -23,3 +23,4 @@ weight: 2
 - **&#x2610; Initial device configuration:** This planned feature aims to provide users with a mechanism to perform the initial configuration of devices during the provisioning process.
```
Comment (jkralik) [inline content/en/docs/services/device-provisioning-service/features.md:23]: also remove this line pls

File: content/en/docs/services/device-provisioning-service/features.md [hunk 3]
```diff
@@ -23,3 +23,4 @@ weight: 2
...
 - **&#x2610; Blacklisting and whitelisting devices:** With this feature, users will have the ability to blacklist or whitelist specific devices for enhanced access control.
 - **&#x2610; Manual approval for device configuration:** This planned feature enables a mechanism where devices require manual approval from the user in order to receive configuration settings. Instead of automatic configuration, users will have control over granting permission for device configuration.
+- **&#x2610; Verify Common Name:** Verify that the Common Name of DPS endpoint certificate matches the name of the selected endpoint.
```
Comment (jkralik) [inline content/en/docs/services/device-provisioning-service/features.md:26]: This is needed in dps-client not in dps-service:
Verify Common Name:** Verify that the Common Name of DPS endpoint certificate matches the name of the selected endpoint by dps client.

File: content/en/docs/services/device-provisioning-service/retry-mechanism.md [hunk 4]
```diff
@@ -29,6 +29,8 @@ The configuration can be changed by the `plgd_dps_set_retry_configuration` funct
 
 Provisioning consists of 3 main steps:
```
Comment (jkralik) [inline content/en/docs/services/device-provisioning-service/retry-mechanism.md:30]: 5 steps :)

File: content/en/docs/services/device-provisioning-service/retry-mechanism.md [hunk 5]
```diff
@@ -29,6 +29,8 @@ The configuration can be changed by the `plgd_dps_set_retry_configuration` funct
...
 
+* synchronization of time
+* requesting and set of device owner
 * sending of signing certificate request
 * requesting and applying of ACLs
 * requesting and applying of plgd hub connection configuration
```
Comment (jkralik) [inline content/en/docs/services/device-provisioning-service/retry-mechanism.md:36]: move up before signing certificate.

File: content/en/docs/services/device-provisioning-service/features.md (hunk 2 above)
Comment (Danielius1922) [reply content/en/docs/services/device-provisioning-service/features.md]: Removed

File: content/en/docs/services/device-provisioning-service/features.md (hunk 3 above)
Comment (Danielius1922) [reply content/en/docs/services/device-provisioning-service/features.md]: Updated

File: content/en/docs/services/device-provisioning-service/retry-mechanism.md (hunk 4 above)
Comment (Danielius1922) [reply content/en/docs/services/device-provisioning-service/retry-mechanism.md]: Fixed

File: content/en/docs/services/device-provisioning-service/retry-mechanism.md (hunk 5 above)
Comment (Danielius1922) [reply content/en/docs/services/device-provisioning-service/retry-mechanism.md]: Fixed

**Review by jkralik — APPROVED** [review]

**Pull request merged**